background.js

```diff
@@ -174,195 +176,214 @@ function backgroundAccept(sockInfo) {
 	}
 }
 
-chrome.runtime.onSuspend.addListener( function(evt) {
-    //createNotification("onSuspend")
-    console.warn('onSuspend')
-    return
+chrome.runtime.onSuspend.addListener(function (evt) {
+	//createNotification("onSuspend")
+	console.warn('onSuspend')
+	return
 
 	// using a persistent socket now...
-    if (localOptions.optBackground) {
-        triggerKeepAwake()
-    } else {
-        if (window.app) app.stop('onsuspend')
-    }
+	if (localOptions.optBackground) {
+		triggerKeepAwake()
+	} else {
+		if (window.app) app.stop('onsuspend')
+	}
 })
-chrome.runtime.onSuspendCanceled.addListener( function(evt) {
-    //createNotification("suspendcanceled!")
-    console.warn('onSuspendCanceled')
+chrome.runtime.onSuspendCanceled.addListener(function (evt) {
+	//createNotification("suspendcanceled!")
+	console.warn('onSuspendCanceled')
 })
 
 function launch(launchData) {
 	HADEVENT = true
 
-    launchData = launchData || {}
-    //if (launchData.source == 'reload') { console.log('app was reloaded'); return }
-    if (launchData.source == 'restart') { console.log('chrome restarted'); return }
-
-    //console.log('onLaunched with launchdata',launchData)
-
-    var info = {type:'onLaunched',
-                launchData: launchData}
-    var opts = {id:'index',
-				outerBounds: { width: 410,
-							   height: 700 }
-			   }
-    //var page = 'index.html'
-    var page = 'browser.html';
-    
-    
-    //launch our app directly here
-    
-      var screenWidth = screen.availWidth;
-      var screenHeight = screen.availHeight;
-      var width = screenWidth;
-      var height = screenHeight;
-      
-      chrome.app.window.create(page, 
-      {
-        outerBounds: {
-        width: width,
-        height: height,
-        
-        }
-      },
-      function(mainWindow) {
-                                 window.mainWindow = mainWindow;
-                                 mainWindow.onClosed.addListener( window_closed )
-                                 var hiddenwin = chrome.app.window.get('hidden')
-                                 if (hiddenwin) { hiddenwin.close() }
-                                 
-			 });
-    
-
-    if (window.app) { console.log('already have webapp',app); return }
+	launchData = launchData || {}
+	//if (launchData.source == 'reload') { console.log('app was reloaded'); return }
+	if (launchData.source == 'restart') { console.log('chrome restarted'); return }
+
+	//console.log('onLaunched with launchdata',launchData)
+
+	// If launched via a url_handler, we need to grab data passed through the URL.
+	// We only want the data that comes after the base URL, specified by the ID
+	// of the url_handler.
+	var relativeUrl;
+	if (launchData.source === 'url_handler' && launchData.url && launchData.id) {
```

Comment:
If the app was launched via a URL handler, that info will be passed in the event parameters. The naming convention used for the `id` properties in the `url_handlers` section of `manifest.json` matches the URL used to launch the app, allowing us to remove the unnecessary information from the beginning of the URL and leave just the portion used to pass parameters.